lib/aida/bot.ex

```diff
@@ -26,28 +38,60 @@ defmodule Aida.Bot do
 
   @spec init(bot :: t) :: {:ok, t}
   def init(bot) do
-    skills = bot.skills
-      |> Enum.map(fn(skill) ->
+    skills =
+      bot.skills
+      |> Enum.map(fn skill ->
         Skill.init(skill, bot)
       end)
 
     {:ok, %{bot | skills: skills}}
   end
 
-  @spec wake_up(bot :: t, skill_id :: String.t, data :: nil | String.t) :: :ok
+  @spec wake_up(bot :: t, skill_id :: String.t(), data :: nil | String.t()) :: :ok
   def wake_up(%Bot{} = bot, skill_id, data \\ nil) do
     find_skill(bot, skill_id)
-      |> Skill.wake_up(bot, data)
+    |> Skill.wake_up(bot, data)
   end
 
-  @spec chat(bot :: t, message :: Message.t) :: Message.t
-  def chat(%Bot{} = bot, %Message{} = message) do
+  @spec chat(bot :: t, message :: Message.t()) :: Message.t()
+  def chat(bot, %Message{} = message) do
     message
     |> reset_language_if_invalid(bot)
     |> choose_language_for_single_language_bot(bot)
     |> handle(bot)
     |> greet_if_no_response_and_language_was_set(bot, message)
     |> unset_session_new
+    |> log(bot)
+  end
+
+  defp log(message, %{public_keys: public_keys} = bot) do
+    content = if message.sensitive do
+      message |> Message.raw() |> Crypto.encrypt(public_keys) |> Poison.encode!
+    else
+      message |> Message.raw()
+    end
+
+    MessageLog.create(%{
+      bot_id: bot.id,
+      session_id: message.session.id,
+      session_uuid: message.session.uuid,
+      content: content,
+      content_type: Atom.to_string(message |> Message.type()),
```

Comment:
While reading back the message logs, how do we know it must be decrypted? Maybe we need to set the `content_type` to `encrypted`, but we need to store the original content type somewhere. We need to rethink this.